src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -61,9 +70,29 @@ let selectionInformation : ?mixed = null;
 
 var DOMRenderer = ReactFiberReconciler({
 
-  getChildHostContext(parentHostContext : string | null, type : string) {
-    const parentNamespace = parentHostContext;
-    return getChildNamespace(parentNamespace, type);
+  getRootHostContext(rootContainerInstance) {
+    const type = rootContainerInstance.tagName.toLowerCase();
+    if (__DEV__) {
+      const namespace = getChildNamespace(null, type);
+      const isMountingIntoDocument = rootContainerInstance.ownerDocument.documentElement === rootContainerInstance;
+      const ancestorInfo = getChildAncestorInfo(null, isMountingIntoDocument ? '#document' : type);
+      return {namespace, ancestorInfo};
+    } else {
```

Comment:
Nit: We don't really use `else` with `DEV` blocks because it didn't use to work in our transforms but I actually prefer that pattern because it enforces the notion that `__DEV__` is only additional and doesn't change behavior. For this case I tend to prefer to use an early return instead. Although in this case I guess it severely changes the whole return type.